internal/testcomponents/example_exporter.go

```diff
@@ -25,81 +25,76 @@ import (
 	"go.opentelemetry.io/collector/pdata/ptrace"
 )
 
-// ExampleExporter is for testing purposes. We are defining an example config and factory
-// for "exampleexporter" exporter type.
-type ExampleExporter struct {
-	config.ExporterSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
-}
-
 const expType = "exampleexporter"
 
-// ExampleExporterFactory is factory for ExampleExporter.
+// ExampleExporterFactory is factory for ExampleExporterConfig.
 var ExampleExporterFactory = component.NewExporterFactory(
 	expType,
 	createExporterDefaultConfig,
 	component.WithTracesExporter(createTracesExporter),
 	component.WithMetricsExporter(createMetricsExporter),
 	component.WithLogsExporter(createLogsExporter))
 
-// CreateDefaultConfig creates the default configuration for the Exporter.
 func createExporterDefaultConfig() config.Exporter {
-	return &ExampleExporter{
+	return &struct {
+		config.ExporterSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
+	}{
```

Comment:
This looks different from what we do with real components. Perhaps for the example to be closer to the reality keep the ExampleExporterConfig struct declared at the top lexical level.

Reply:
This is not an example despite the name, it is used for testing. But will update to follow the standard.